### PR TITLE
Use py.test ini file integration instead of rolling our own

### PR DESCRIPTION
**Pull request by davehunt**
* Also changes `privacy` tag to be clearly related to Sauce Labs job visibility.
* Also migrates the Sauce Labs and BrowserStack documentation.

### CODE REVIEW

**Comment by davehunt** [comment]
Pinging @bobsilverberg for review.

File: pytest_selenium/cloud/saucelabs.py [hunk 1]
```diff
@@ -77,12 +62,13 @@ def start_driver(item, options, capabilities):
...
         capabilities['version'] = options.browser_version
-    executor = 'http://%s:%s@ondemand.saucelabs.com:80/wd/hub' % _credentials()
+    executor = 'http://%s:%s@ondemand.saucelabs.com:80/wd/hub' % \
+        _credentials(item.config)
     return webdriver.Remote(command_executor=executor,
                             desired_capabilities=capabilities)
 
 
-def url(session):
+def url(config, session):
```
Comment (bobsilverberg) [inline pytest_selenium/cloud/saucelabs.py]: `config` isn't used in this `url` method.

File: pytest_selenium/pytest_selenium.py [hunk 2]
```diff
@@ -110,11 +110,6 @@ def pytest_runtest_makereport(__multicall__, item, call):
     extra_summary = []
     report = __multicall__.execute()
     extra = getattr(report, 'extra', [])
-    try:
-        report.public = item.keywords['privacy'].args[0] == 'public'
```
Comment (bobsilverberg) [inline pytest_selenium/pytest_selenium.py]: I see `report.public` being removed, but not replaced with anything. Was it not actually being used for anything before?

File: pytest_selenium/cloud/saucelabs.py (hunk 1 above)
Comment (davehunt) [inline pytest_selenium/cloud/saucelabs.py]: No, but it's passed to it.

Dave Hunt

File: pytest_selenium/pytest_selenium.py [hunk 3]
```diff
@@ -136,29 +131,50 @@ def pytest_runtest_makereport(__multicall__, item, call):
...
 
 
 def pytest_addoption(parser):
-    config = ConfigParser.ConfigParser(defaults={'baseurl': ''})
-    config.read('mozwebqa.cfg')
+    parser.addini('selenium_base_url', 'base url for selenium')
+
+    # browserstack configuration
+    parser.addini('browserstack_username',
+                  'browserstack username',
```
Comment (bobsilverberg) [inline pytest_selenium/pytest_selenium.py]: Below (for sauce labs config) you include the `help=` argument name, which makes it clearer what the second argument is, but here you leave it out. I think for consistency and clarity you should add it here as well.

File: pytest_selenium/pytest_selenium.py (hunk 2 above)
Comment (davehunt) [inline pytest_selenium/pytest_selenium.py]: Yeah, it was an artefact from the log publicity checking of RC.

Dave Hunt

File: pytest_selenium/pytest_selenium.py (hunk 3 above)
Comment (davehunt) [inline pytest_selenium/pytest_selenium.py]: +1 thanks.

Dave Hunt

**Comment by bobsilverberg** [comment]
Very nice @davehunt! Great to be able to use code built into py.test instead of writing our own to parse the config file.

r+ wc

**Comment by davehunt** [comment]
Merged in https://github.com/davehunt/pytest-selenium/commit/6cdd173fdde851b5c9171370fa99851e52731682

**Pull request closed without merging**